File: demo/src/main/java/Borala/app/demo/Model/Subscription.java
package Borala.app.demo.Model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "subscriptions")
@Getter
@Setter
public class Subscription {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String email;
    private String phone;

    @ManyToOne
    @JoinColumn(name = "event_id", nullable = false)
    private Event event;
}

File: lib/models/subscription.dart
class Subscription {
  final String name;
  final String email;
  final String phone;
  final int eventId;

  Subscription({
    required this.name,
    required this.email,
    required this.phone,
    required this.eventId,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'email': email,
      'phone': phone,
      'eventId': eventId,
    };
  }
}
